### DISCUSSION TITLE
Delta 2: Adjust AC Charging Power through Home Assistant REST API has no effect on actual device, only on internal state of the slider

### DISCUSSION THREAD

**Discussion by mussiniinfo** [tolwi/hassio-ecoflow-cloud#97] [General]
I have a Delta 2 and I want to continuously adjust its AC charging power rate, in the allowed range 200-1000 W, to continuously match (and capture) as precisely as possible the production surplus of my solar power generation installation.

I do not want to implement this as an automation within Home Assistant. I have an external dedicated server that runs all necessary algorithms. Maybe it may look a bit unusual but in my context I am using Home Assistant only as a convenient adapter to control a wide variety of things from the "real" automation system, that is different and elsewhere.

I can already read the SoC level of the Delta 2, and the surplus production of the solar array. This would let me react by setting the new proper value for the AC Charging Rate on the Delta 2.

Unfortunately, if I invoke the Home Assistant REST API to set the number value of the slider, on the Home Assistant GUI the slider indeed moves instantaneously, but nothing happens on the device (no confirmation "beep" and no actual change anyway), and of course, nothing happens on the Ecoflow app on connected smartphones, either. 

```
curl -X POST -H "Authorization: Bearer $BEARER" -H "Content-Type: application/json" -d '{"entity_id": "number.r331zeb4zebp0782_ac_charging_power", "state": "$inputpower", "attributes":{"min":200,"max":1200,"step":100,"mode":"slider","unit_of_measurement":"W","device_class":"power","icon":"mdi:transmission-tower-import","friendly_name":"potenza_ricarica"}}' http://<ADDR>:<PORT>/api/states/number.r331zeb4zebp0782_ac_charging_power
```
where $inputpower is the desired value in Watt, and $BEARER is the long-lived auth token.

The effect of the curl on the REST API seems to be limited to updating the internal state of the slider in the Home Assistant web GUI, with NO triggered SET action of the associated parameter on the Delta 2.
This is strange because of course if I manually move the AC Charging Power slider on the Home Assistant Web UI, the device DOES receive the update, and the app, after a refresh, DOES reflect the changed value.

What am I doing wrong?

Perhaps there is some REST API with the meaning of "commit to device" that I should call after I have changed the slider position?

Or perhaps my usage pattern is simply not supported by design in this HA integration? I was expecting that, just like from REST API I can read parameter values that actually come from the device, I was also able to write parameters to actual device, also through the same REST API. This does not seem the case (unless I am missing something).

In general, does anyone know a way to programmatically set from outside Home Assistant, through its REST API, a parameter of the Delta 2 device, WITH actual effect on the device and not just on internal state of a GUI slider?

**Comment by tolwi** [comment]
Try the following example
```
curl \
  -H "Authorization: Bearer *******" \
  -H "Content-Type: application/json" \
  -d '{"entity_id": "number.delta2_ac_charging_power", "value": 600}' \
  http://localhost:8123/api/services/number/set_value
```
